lms/product/product.py

```diff
@@ -0,0 +1,47 @@
+"""Core models of the product."""
+
+from dataclasses import dataclass
+from enum import Enum
+
+from lms.services.grouping.plugin import GroupingServicePlugin
+
+
+class Family(str, Enum):
+    """Enum for which product this relates to."""
+
+    BLACKBAUD = "BlackbaudK12"
+    BLACKBOARD = "BlackboardLearn"
+    CANVAS = "canvas"
+    D2L = "desire2learn"
+    MOODLE = "moodle"
+    SAKAI = "sakai"
+    SCHOOLOGY = "schoology"
+    UNKNOWN = "unknown"
+
+    @classmethod
+    def _missing_(cls, _value):
+        return cls.UNKNOWN
+
+
+@dataclass
+class Plugins:
+    """A collection of plugins used to separate LMS specific functionality."""
+
+    grouping_service: GroupingServicePlugin
+
+
+@dataclass
+class Product:
```

Comment:
I'd personally collapse `Product` and `GenericProduct` into just one class. I'm not sure the extra layer is giving us much.

Reply:
Yeah... it could be. My heart is telling me not to though. Not sure why.

As it is, the `GenericProduct` almost exactly mirrors the kind of thing you have to implement. And I like the total separation between the interface and the implementation.